mesonbuild/dependencies/base.py

```diff
@@ -813,27 +823,25 @@ def get_methods():
         return [DependencyMethods.PKGCONFIG]
 
     def check_pkgconfig(self, pkgbin):
-        pkgbin = ExternalProgram(pkgbin, silent=True)
-        if pkgbin.found():
-            try:
-                p, out = Popen_safe(pkgbin.get_command() + ['--version'])[0:2]
-                if p.returncode != 0:
-                    mlog.warning('Found pkg-config {!r} but couldn\'t run it'
-                                 ''.format(' '.join(pkgbin.get_command())))
-                    # Set to False instead of None to signify that we've already
-                    # searched for it and not found it
-                    pkgbin = False
-            except (FileNotFoundError, PermissionError):
-                pkgbin = False
-        else:
-            pkgbin = False
-        if not self.silent:
-            if pkgbin:
-                mlog.log(mlog.bold('pkg-config'), 'found:', mlog.green('YES'), '({})'.format(pkgbin.get_path()),
-                         out.strip())
-            else:
-                mlog.log(mlog.bold('pkg-config'), 'found:', mlog.red('NO'))
-        return pkgbin
+        if not pkgbin.found():
+            mlog.log('Did not find anything at {!r}'
+                     ''.format(' '.join(pkgbin.get_command())))
+            return None
+        try:
+            p, out = Popen_safe(pkgbin.get_command() + ['--version'])[0:2]
+            if p.returncode != 0:
+                mlog.warning('Found pkg-config {!r} but it failed when run'
+                             ''.format(' '.join(pkgbin.get_command())))
+                return None
+        except FileNotFoundError:
+            mlog.warning('We thought we found pkg-config {!r} but now it\'s not there. How odd!'
+                         ''.format(' '.join(pkgbin.get_command())))
+            return None
+        except PermissionError:
+            mlog.warning('Found pkg-config {!r} but didn\'t have permissions to run it.'
+                         ''.format(' '.join(pkgbin.get_command())))
```

Comment:
It might be good to mention something about the executable bit on nix systems.

Reply:
You mean just in the error message?

Reply:
Yeah, generally on Linux a PermissionError means that you don't have the proper `x` bit

Reply:
OK cool. That's easier than trying to find the python to inspect the permissions and do something more detailed.

Reply:
it's not actually that hard to get the permissions, `os.stat`. But I think it's fine to just have something like:
```python
msg = ...
if not is_windows():
   msg += '\n\nOn Unix-like systems this is often caused by scripts that are not executable.'
```
or something like that.